### PR TITLE
MAYN-281: Update edx-configuration to support multiple theme directories

### PR DESCRIPTION
**Pull request by saleem-latif**
Hi @mattdrayer , @fredsmith , @feanil 

Kindly take a look at this PR, it contains changes for MAYN-281,

**Description of MAYN-281:**
Currently, we can specify only one theme directory for themes. Update configuration so that multiple theme dirs can be specified
Here is a list of changes
1. Rename `ECOMMERCE_COMPREHENSIVE_THEME_DIR` to  `ECOMMERCE_COMPREHENSIVE_THEME_DIRS` and make it accept list of directories instead of a single directory.
2. `COMPREHENSIVE_THEME_DIR` is renamed to `COMPREHENSIVE_THEME_DIRS` in ecommerce, so we need to make corresponding update in ansible scripts too.
3. create a group named `edx-themes` with read/write permissions to all theme directories so that any ansible user (like ecommerce, edxapp) that requires access to edx-themes can join this group for proper access rights.
4. Remove ansible scripts that copy themes from edx-themes dir into ecommerce/themes.

### CODE REVIEW

**Comment by mattdrayer** [comment]
Overall LGTM @saleem-latif -- 👍 and ready for @fredsmith and/or @feanil to review

File: playbooks/roles/edx-themes/tasks/main.yml [hunk 1]
```diff
@@ -25,5 +25,45 @@
...
+
+# make sure edx-theme's group has read/write access to themes directory
+- name: ensure edx-theme's group has read/write access to themes directory
+  file:
+    path: "{{ themes_home }}"
+    state: directory
+    recurse: yes
+    owner: "{{ themes_user }}"
+    group: "{{ themes_group }}"
+    mode: "g+rw"
```
Comment (fredsmith) [inline playbooks/roles/edx-themes/tasks/main.yml]: (yey)

File: playbooks/roles/edx-themes/tasks/main.yml [hunk 2]
```diff
@@ -25,5 +25,45 @@
...
+    recurse: yes
+    group: "{{ themes_group }}"
+    mode: "g+rw"
+  with_items: additional_theme_dirs
+  when: additional_theme_dirs is defined
+  tags:
+    - install
+    - install:base
+
+- name: Add theme users to theme's group so that that have read/erite access to themes directories
```
Comment (fredsmith) [inline playbooks/roles/edx-themes/tasks/main.yml]: in the add-user role, you're creating users without append: yes.  I believe that will cause the group list to be overwritten by that play, depending on ordering of the plays.

I think you should call user with append: yes there as well. What do you think?

Comment (saleem-latif) [inline playbooks/roles/edx-themes/tasks/main.yml]: @fredsmith yes, that sounds good, I will update add-user role also.

Comment (saleem-latif) [inline playbooks/roles/edx-themes/tasks/main.yml]: @fredsmith added `append: yes` on user creation in add-user role, kindly take a look

**Comment by saleem-latif** [comment]
@fredsmith added append: yes on user creation in add-user role, kindly take a look

**Comment by fredsmith** [comment]
travis timed out.  Have you built a sandbox with this branch @saleem-latif ?

**Comment by saleem-latif** [comment]
@fredsmith yes, sandbox with these configuration changes and ecommerce [PR#749](https://github.com/edx/ecommerce/pull/749) is 'multi-theme-dirs'
Here are links with different themes applied to the sandbox.
1. https://payments-red-theme.sandbox.edx.org/ (Red theme applied)
2. https://payments-mit-theme.sandbox.edx.org (MIT Theme applied)
3. https://ecommerce-multi-theme-dirs.sandbox.edx.org (default edx.org theme applied)
4. https://studio-multi-theme-dirs.sandbox.edx.org (sandbox studio)

Also, since this PR is dependent on the changes for WL-451 ,that is why it is showing conflicts, I will rebase it with master once [PR#3071](https://github.com/edx/configuration/pull/3071) is merged.

**Comment by fredsmith** [comment]
LGTM - I'd like a second 👍 before we merge, though.  @edx/devops 

@jibsheet - this may help resolve some of the build issues we've been running into by removing the rsync task.

**Comment by mattdrayer** [comment]
I'd also like to confirm there is no impact to edx-platform theming with all of these changes

**Comment by jibsheet** [comment]
I'll try to review later today, but I think both this and the other reverted branch which broke a build should follow our standard documented "large ansible testing" process.

https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist

especially since the failure was totally silent, I wouldn't merge either branch without a known AMI build passing.

**Comment by mattdrayer** [comment]
@saleem-latif looks like this branch is in need of a rebase

**Comment by saleem-latif** [comment]
@mattdrayer it will be rebased after the merge of [PR#3071](https://github.com/edx/configuration/pull/3071)

File: playbooks/roles/edx_themes/defaults/main.yml [hunk 3]
```diff
@@ -21,7 +23,7 @@ THEMES_GIT_PROTOCOL: "{{ COMMON_GIT_PROTOCOL }}"
 THEMES_GIT_MIRROR: "{{ COMMON_GIT_MIRROR }}"
 THEMES_GIT_PATH: "{{ COMMON_GIT_PATH }}"
 THEMES_REPO: "sample-themes.git"
-THEMES_VERSION: "master"
+THEMES_VERSION: "release"
```
Comment (mattdrayer) [inline playbooks/roles/edx_themes/defaults/main.yml:26]: @douglashall FYI -- when this PR merges we will need to begin using a release branch for edx-themes

File: playbooks/roles/add_user/tasks/main.yml [hunk 4]
```diff
@@ -15,7 +15,18 @@
 #
 # This role performs the repetitive tasks that most edX roles
 # require in our default configuration.
-#
+
+# create groups for the user
```
Comment (feanil) [inline playbooks/roles/add_user/tasks/main.yml:19]: The group with the same name as the user gets created by default.  Is there a scenario that you ran into where that is not true?

File: playbooks/roles/ecommerce/defaults/main.yml [hunk 5]
```diff
@@ -115,7 +115,7 @@ ECOMMERCE_THEME_SCSS: 'sass/themes/default.scss'
 # Directory name inside edx-themes repo that contain ecommerce themes
 ECOMMERCE_THEMES_DIR_NAME: 'ecommerce'
 
-ECOMMERCE_COMPREHENSIVE_THEME_DIR: !!null
+ECOMMERCE_COMPREHENSIVE_THEME_DIRS: !!null
```
Comment (feanil) [inline playbooks/roles/ecommerce/defaults/main.yml:118]: This is a backwards incompatible change, we should have a changelog entry.  Also, the secure config will need to be updated before we can build an AMI for this, is there a ticket for that work?

File: playbooks/roles/add_user/tasks/main.yml (hunk 4 above)
Comment (saleem-latif) [inline playbooks/roles/add_user/tasks/main.yml:19]: @feanil yes it was failing for analytics api, here is the stack trace

```
08:27:58 TASK: [analytics_api | setup the analytics_api env file] **********************

08:27:58 failed: [ansibletest-scratch.sandbox.edx.org] => {"failed": true, "gid": 0, "group": "root", "mode": "0644", "owner": "analytics_api", "path": "/edx/app/analytics_api/analytics_api_env", "size": 193, "state": "file", "uid": 1011}

08:27:58 msg: chgrp failed: failed to look up group analytics_api

08:28:00 FATAL: all hosts have already failed -- aborting
```

File: playbooks/roles/ecommerce/defaults/main.yml (hunk 5 above)
Comment (saleem-latif) [inline playbooks/roles/ecommerce/defaults/main.yml:118]: @feanil [DEVOPS-4309](https://openedx.atlassian.net/browse/DEVOPS-4309) is the ticket to track secure configuration changes.

**Comment by mattdrayer** [comment]
@saleem-latif -- please add a CHANGELOG.md entry as requested by @feanil -- otherwise @feanil are we good to move forward with this PR?

**Comment by saleem-latif** [comment]
@mattdrayer , @feanil added `CHANGELOG.md` entry

**Comment by fredsmith** [comment]
https://github.com/edx-ops/edx-secure/pull/894 is ready to merge to handle the secure changes for ecommerce.

File: playbooks/roles/ecommerce/defaults/main.yml (hunk 5 above)
Comment (fredsmith) [inline playbooks/roles/ecommerce/defaults/main.yml:118]: https://github.com/edx-ops/edx-secure/pull/894

**Comment by feanil** [comment]
Cool, I think we should build an AMI from this for edxapp in the loadtest environment to make sure this doesn't break that process since we've had issues in the past where sandboxes have worked but AMIs didn't build correctly.  Once that is verified. :+1:

**Comment by fredsmith** [comment]
loadtest build succeeded 👍

**Pull request merged**